File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        res = []
        arr = sorted(arr)
        minDiff = float('inf')
        for i in range(len(arr) - 1):
            minDiff = min(minDiff, arr[i + 1] - arr[i])
        for i in range(len(arr) - 1):
            if arr[i + 1] - arr[i] == minDiff:
                res.append([arr[i], arr[i + 1]])
        return res

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        vector<vector<int>> res;
        int minDiff = INT_MAX;
        sort(arr.begin(), arr.end());
        for(int i = 0; i < arr.size() - 1; i++)
        {
            minDiff = min(minDiff, arr[i + 1] - arr[i]);
        }
        for(int i = 0; i < arr.size() - 1; i++)
        {
            if(arr[i + 1] - arr[i] == minDiff)
            {
                res.push_back({arr[i], arr[i + 1]});
            }
        }
        return res;
    }
};